flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/SubtaskCheckpointCoordinatorImpl.java

```diff
@@ -175,6 +203,77 @@ public void checkpointState(
 		}
 	}
 
+	private OperatorSnapshotFutures buildOperatorSnapshotFutures(
+			CheckpointMetaData checkpointMetaData,
+			CheckpointOptions checkpointOptions,
+			OperatorChain<?, ?> operatorChain,
+			StreamOperator<?> op,
+			Supplier<Boolean> isCanceled,
+			ChannelStateWriteResult channelStateWriteResult) throws Exception {
+		CheckpointStreamFactory storage = checkpointStorage.resolveCheckpointStorageLocation(
+			checkpointMetaData.getCheckpointId(),
+			checkpointOptions.getTargetLocation());
+		OperatorSnapshotFutures snapshotInProgress = checkpointStreamOperator(
+			op,
+			checkpointMetaData,
+			checkpointOptions,
+			storage,
+			isCanceled);
+		if (op == operatorChain.getHeadOperator()) {
+			snapshotInProgress.setInputChannelStateFuture(channelStateWriteResult
+															.getInputChannelStateHandles()
+															.thenApply(StateObjectCollection::new)
+															.thenApply(SnapshotResult::of)
+			);
+		}
+		if (op == operatorChain.getTailOperator()) {
+			snapshotInProgress.setResultSubpartitionStateFuture(channelStateWriteResult
+																	.getResultSubpartitionStateHandles()
+																	.thenApply(StateObjectCollection::new)
+																	.thenApply(SnapshotResult::of)
+			);
+		}
+		return snapshotInProgress;
+	}
+
+	@Override
+	public void close() {
+		channelStateWriter.close();
+	}
+
+	// Caches checkpoint output stream factories to prevent multiple output stream per checkpoint.
```

Comment:
The proper javadoc format for class should be 
/**
*
**/

Reply:
This is not a javadoc but rather description of `SubtaskCheckpointCoordinatorImpl` inner workings. I can make it javadoc if you prefer though.

Reply:
I am not forcing that, it is up to you. Only saw that most of descriptions for class scope prefers to using above, and `//` is almost using inline codes, `/** */` is for class fields.